operator/v1/stable.ingresscontroller.testsuite.yaml

```diff
@@ -13,3 +13,248 @@ tests:
+  - name: Should not allow to set/delete dynamic response headers values with not allowed sample fetchers.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-not-allowed-values
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY      
+            - name: X-Target 
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdrs(host),lower]"
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are res.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should not allow to set/delete dynamic response headers values with not allowed converters.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-not-allowed-values
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY      
+            - name: X-Source 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,bogus]"
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are res.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should not allow to set/delete dynamic response headers values contains sample fetcher req.hdr.
```

Comment:
Counterpart for the request header not allowing the `res.hdr` fetcher?

Reply:
yes res.hdr needs to be in http-response

Reply:
We need a test to show that you can't set the `res.hdr` in the request regex